containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceListener.java

```diff
@@ -70,7 +71,61 @@ public void stop() {
      * @return The id of the namespace or null if the namespace does not exist
      */
     public String getNamespaceId(String namespaceName) {
-        return namespaces.get(namespaceName);
+        String namespaceId = namespaces.get(namespaceName);
```

Comment:
``computeIfAbsent`` would reduce possible multiple queries to the Kubernetes master,

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-